app/code/Magento/ConfigurableProduct/Pricing/Price/ConfigurablePriceResolver.php

```diff
@@ -67,6 +75,88 @@ public function resolvePrice(\Magento\Framework\Pricing\SaleableInterface $produ
             $price = isset($price) ? min($price, $productPrice) : $productPrice;
         }
 
+        if ($this->hasTierPrice($product)) {
+            $tierPrice = $this->getMinimalTierPrice($product);
+            $price = isset($price) ? min($price, $tierPrice) : $price;
+        }
+
         return (float)$price;
     }
+
+    /**
+     * Check if at least one child product has tier price
+     *
+     * @param SaleableInterface|Product $product
+     * @return bool
+     */
+    private function hasTierPrice(SaleableInterface $product)
```

Comment:
Please, provide the return types where it's possible. It applies also for other methods 
```suggestion
    private function hasTierPrice(SaleableInterface $product): bool
```